ehrql/query_engines/base_sql.py

```diff
@@ -634,6 +634,14 @@ def get_results(self, variable_definitions):
         results_query = self.get_query(variable_definitions)
         setup_queries, cleanup_queries = get_setup_and_cleanup_queries(results_query)
         with self.engine.connect() as connection:
+
+            def do_cleanup():
+                for i, cleanup_query in enumerate(cleanup_queries, start=1):
+                    log.info(
+                        f"Running cleanup query {i:03} / {len(cleanup_queries):03}"
+                    )
+                    connection.execute(cleanup_query)
```

Comment:
I can understand exactly why you made this change, but I'm not convinced it's the right approach here. There are any number of ways we could still fail to clean up e.g. failing during one of the setup queries, or a failure which leaves the connection dead. So we're going to need some out-of-band mechanism to clean these things up in any case. I think it would leave our code easier to reason about if we just exited on error and didn't attempt to clean up.

Reply:
OK - I think the global unique ID for the table might help with this. One of the issues was that if the generative tests use an inline table and encounter one of the ignored errors, the inline table didn't get cleaned up and other gen tests would fail. Although if it's in the same test run, it might still be a problem. I'll try it.